nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/RealmsManagerHolder.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022, 2023 Contributors to the Eclipse Foundation.
+ * Copyright (c) 1997, 2021 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package com.sun.enterprise.security.auth.realm;
+
+import java.lang.ref.WeakReference;
+
+import org.glassfish.internal.api.Globals;
+
+public class RealmsManagerHolder {
+
+    private static WeakReference<RealmsManager> realmsManagerReference = new WeakReference<RealmsManager>(null);
```

Comment:
@arjantijms, `realmsManagerReferense` must be volatile to ensure visibility by other threads. Or method `getRealmsMnager()` must be synchronized instead of `_getRealmsManager()`.

Reply:
Good point, although this is something that was already in the existing code. I just moved it to another class (which is why I also kept the original Oracle copyright)